tests/Valkey.Glide.IntegrationTests/GeospatialCommandTests.cs

```diff
@@ -0,0 +1,974 @@
+    {
+        string key = Guid.NewGuid().ToString();
+        GeoEntry[] entries =
+        [
+            new GeoEntry(13.361389, 38.115556, "Palermo"),
+            new GeoEntry(15.087269, 37.502669, "Catania")
+        ];
+        await client.GeoAddAsync(key, entries);
+        
+        // Test all units with expected values (approximate distance between Palermo and Catania)
+        double? distanceMeters = await client.GeoDistanceAsync(key, "Palermo", "Catania", GeoUnit.Meters);
+        double? distanceKilometers = await client.GeoDistanceAsync(key, "Palermo", "Catania", GeoUnit.Kilometers);
+        double? distanceMiles = await client.GeoDistanceAsync(key, "Palermo", "Catania", GeoUnit.Miles);
+        double? distanceFeet = await client.GeoDistanceAsync(key, "Palermo", "Catania", GeoUnit.Feet);
+        
+        // Verify all distances are returned
+        Assert.NotNull(distanceMeters);
+        Assert.NotNull(distanceKilometers);
+        Assert.NotNull(distanceMiles);
+        Assert.NotNull(distanceFeet);
+        
+        // Verify approximate expected values (distance between Palermo and Catania)
+        Assert.True(distanceMeters > 166000 && distanceMeters < 167000); // ~166274 meters
+        Assert.True(distanceKilometers > 166 && distanceKilometers < 167); // ~166.27 km
+        Assert.True(distanceMiles > 103 && distanceMiles < 104); // ~103.31 miles
+        Assert.True(distanceFeet > 545000 && distanceFeet < 546000); // ~545,518 feet
+        
+        // Verify unit conversions are consistent
+        double metersToKm = distanceMeters.Value / 1000;
+        double metersToMiles = distanceMeters.Value / 1609.344;
+        double metersToFeet = distanceMeters.Value * 3.28084;
+        
+        Assert.True(Math.Abs(metersToKm - distanceKilometers.Value) < 0.001);
+        Assert.True(Math.Abs(metersToMiles - distanceMiles.Value) < 0.001);
+        Assert.True(Math.Abs(metersToFeet - distanceFeet.Value) < 1); // Allow 1 foot tolerance
+    }
+    
+    [Theory(DisableDiscoveryEnumeration = true)]
+    [MemberData(nameof(Config.TestClients), MemberType = typeof(TestConfiguration))]
+    public async Task GeoDistance_DefaultUnit_ReturnsMeters(BaseClient client)
+    {
+        string key = Guid.NewGuid().ToString();
+        GeoEntry[] entries =
+        [
+            new GeoEntry(13.361389, 38.115556, "Palermo"),
+            new GeoEntry(15.087269, 37.502669, "Catania")
+        ];
+        await client.GeoAddAsync(key, entries);
+        
+        // Test default unit (should be meters)
+        double? distanceDefault = await client.GeoDistanceAsync(key, "Palermo", "Catania");
+        double? distanceMeters = await client.GeoDistanceAsync(key, "Palermo", "Catania", GeoUnit.Meters);
+        
+        Assert.NotNull(distanceDefault);
+        Assert.NotNull(distanceMeters);
+        Assert.Equal(distanceMeters.Value, distanceDefault.Value, 1e-9);
```

Comment:
Is this considered best practice in C#? Won't the test fail even if we just have the `AssertEqual` test if one of the value is `null`?

```suggestion
        Assert.Equal(distanceMeters.Value, distanceDefault.Value, 1e-9);
```